pingcap/tidb-operator
# set tcp_keepalive_time for tidb on aliyun

### What problem does this PR solve? 
Set tcp_keepalive_time for tidb on Ali ACK, part of https://github.com/pingcap/tidb-operator/issues/880
### What is changed and how does it work?
Set podSecurityContext for TiDB.
In terraform, enable configuration of net.* sysctls in kubelet args.
### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
    - Deploy

      Before this PR:
      ```
      Pod:
      [root@248 aliyun]$kubectl  --kubeconfig credentials/kubeconfig -n       my-cluster exec -it my-cluster-tidb-0 -c tidb -- sh
      / # sysctl -a | grep keepalive
      net.ipv4.tcp_keepalive_intvl = 75
      net.ipv4.tcp_keepalive_probes = 9
      net.ipv4.tcp_keepalive_time = 7200
      
      Kubelet:
      [root@z ~]# ps -ef | grep kubelet
      root      5782     1  2 14:01 ?        00:02:58 /usr/bin/kubelet       --bootstrap-kubeconfig=...
      
      MySQL:
      MySQL [(none)]> select sleep(960); select tidb_version(); 
      ERROR 2013 (HY000): Lost connection to MySQL server during query
      ERROR 2006 (HY000): MySQL server has gone away
      No connection. Trying to reconnect...
      Connection id:    9639
      Current database: *** NONE ***
      
      +-------------------------------------------------------------------------------------      -------------------------------------------------------------------------------------      -------------------------------------------------------------------------------------      ------------------------------------------+
      | tidb_version()                                                                                                                                                                                                                                                                                                            |
      +-------------------------------------------------------------------------------------      -------------------------------------------------------------------------------------      -------------------------------------------------------------------------------------      ------------------------------------------+
      | Release Version: v3.0.4
      Git Commit Hash: 694e086e7914a8fc0eb601327edb6bcc31d2c7f2
      Git Branch: HEAD
      UTC Build Time: 2019-10-08 08:13:34
      GoVersion: go version go1.12.10 linux/amd64
      Race Enabled: false
      TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
      Check Table Before Drop: false |
      +-------------------------------------------------------------------------------------      -------------------------------------------------------------------------------------      -------------------------------------------------------------------------------------      ------------------------------------------+
      1 row in set (0.01 sec)
      ```
      After this PR:
      ```
      Pod:
      [root@248 aliyun]$kubectl exec -it my-cluster-tidb-0 -n my-cluster -c tidb       -- sh
      / # sysctl -a | grep keepalive
      sysctl: error reading key 'net.ipv6.conf.all.stable_secret': I/O error
      net.ipv4.tcp_keepalive_intvl = 75
      net.ipv4.tcp_keepalive_probes = 9
      net.ipv4.tcp_keepalive_time = 300
      
      Kubelet:
      [root@x ~]# ps -ef | grep kubelet
      root     27380     1  2 16:27 ?        00:00:07 /usr/bin/kubelet       --bootstrap-kubeconfig=... --allowed-unsafe-sysctls=net.*
      
      MySQL:
      
      MySQL [(none)]> select sleep(960); select tidb_version();
      +------------+
      | sleep(960) |
      +------------+
      |          0 |
      +------------+
      1 row in set (16 min 0.00 sec)
      
      +-------------------------------------------------------------------------------------      -------------------------------------------------------------------------------------      -------------------------------------------------------------------------------------      ------------------------------------------+
      | tidb_version()                                                                                                                                                                                                                                                                                                            |
      +-------------------------------------------------------------------------------------      -------------------------------------------------------------------------------------      -------------------------------------------------------------------------------------      ------------------------------------------+
      | Release Version: v3.0.4
      Git Commit Hash: 694e086e7914a8fc0eb601327edb6bcc31d2c7f2
      Git Branch: HEAD
      UTC Build Time: 2019-10-08 08:13:34
      GoVersion: go version go1.12.10 linux/amd64
      Race Enabled: false
      TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
      Check Table Before Drop: false |
      +-------------------------------------------------------------------------------------      -------------------------------------------------------------------------------------      -------------------------------------------------------------------------------------      ------------------------------------------+
      1 row in set (0.01 sec)
      
      MySQL [(none)]> 

      ```
   - Reboot ECS

      tcp_keepalive_time keeps working.
   - Recover ECS by Auto Scaling Group

      tcp_keepalive_time keeps working on new ECS.
   - Scale out TiDB

      tcp_keepalive_time keeps working on new ECS & TiDB.
   - Scale in TiDB
   
      tcp_keepalive_time keeps working on remaining ECS & TiDB.
   - ACK Upgrade 
      Currently, only two versions during ACK cluster creation, 1.12 & 1.14, but upgrade from 1.12 to 1.14 is not supported yet on ACK.

Code changes

 - Has Terraform scripts change

### Does this PR introduce a user-facing change?:
 
 ```release-note
Set tcp_keepalive_time to 300s for tidb on aliyun.
 ```
